src/main/java/com/yahoo/sketches/tuple/DirectArrayOfDoublesQuickSelectSketch.java

```diff
@@ -0,0 +1,328 @@
+    mem_.putByte(FLAGS_BYTE, (byte) (
+      (isBigEndian ? 1 << Flags.IS_BIG_ENDIAN.ordinal() : 0) |
+      (samplingProbability < 1f ? 1 << Flags.IS_IN_SAMPLING_MODE.ordinal() : 0) |
+      (1 << Flags.IS_EMPTY.ordinal())
+    ));
+    mem_.putByte(NUM_VALUES_BYTE, (byte) numValues);
+    mem_.putShort(SEED_HASH_SHORT, Util.computeSeedHash(seed));
+    theta_ = (long) (Long.MAX_VALUE * (double) samplingProbability);
+    mem_.putLong(THETA_LONG, theta_);
+    mem_.putByte(LG_NOM_ENTRIES_BYTE, (byte) Integer.numberOfTrailingZeros(nomEntries));
+    mem_.putByte(LG_CUR_CAPACITY_BYTE, (byte) Integer.numberOfTrailingZeros(startingCapacity));
+    mem_.putByte(LG_RESIZE_FACTOR_BYTE, (byte) lgResizeFactor);
+    mem_.putFloat(SAMPLING_P_FLOAT, samplingProbability);
+    mem_.putInt(RETAINED_ENTRIES_INT, 0);
+    keysOffset_ = ENTRIES_START;
+    valuesOffset_ = keysOffset_ + SIZE_OF_KEY_BYTES * startingCapacity;
+    mem_.clear(keysOffset_, SIZE_OF_KEY_BYTES * startingCapacity); // clear keys only
+    mask_ = startingCapacity - 1;
+    lgCurrentCapacity_ = Integer.numberOfTrailingZeros(startingCapacity);
+    setRebuildThreshold();
+  }
+
+  /**
+   * Wraps the given Memory assuming the default update seed
+   * @param mem <a href="{@docRoot}/resources/dictionary.html#mem">See Memory</a>
+   */
+  public DirectArrayOfDoublesQuickSelectSketch(Memory mem) {
+    this(mem, DEFAULT_UPDATE_SEED);
+  }
+
+  /**
+   * Wraps the given Memory.
+   * @param mem <a href="{@docRoot}/resources/dictionary.html#mem">See Memory</a>
+   * @param seed update seed
+   */
+  public DirectArrayOfDoublesQuickSelectSketch(Memory mem, long seed) {
+    super(mem.getByte(NUM_VALUES_BYTE), seed);
+    mem_ = mem;
+    SerializerDeserializer.validateFamily(mem.getByte(FAMILY_ID_BYTE), mem.getByte(PREAMBLE_LONGS_BYTE));
+    SerializerDeserializer.validateType(mem_.getByte(SKETCH_TYPE_BYTE), SerializerDeserializer.SketchType.ArrayOfDoublesQuickSelectSketch);
+    byte version = mem_.getByte(SERIAL_VERSION_BYTE);
+    if (version != serialVersionUID) throw new RuntimeException("Serial version mismatch. Expected: " + serialVersionUID + ", actual: " + version);
+    boolean isBigEndian = mem.isAllBitsSet(FLAGS_BYTE, (byte) (1 << Flags.IS_BIG_ENDIAN.ordinal()));
+    if (isBigEndian ^ ByteOrder.nativeOrder().equals(ByteOrder.BIG_ENDIAN)) throw new RuntimeException("Byte order mismatch");
+    Util.checkSeedHashes(mem.getShort(SEED_HASH_SHORT), Util.computeSeedHash(seed));
+    keysOffset_ = ENTRIES_START;
+    valuesOffset_ = keysOffset_ + SIZE_OF_KEY_BYTES * getCurrentCapacity();
+    // to do: make parent take care of its own parts
+    mask_ = getCurrentCapacity() - 1;
+    lgCurrentCapacity_ = Integer.numberOfTrailingZeros(getCurrentCapacity());
+    theta_ = mem_.getLong(THETA_LONG);
+    isEmpty_ = mem_.isAllBitsSet(FLAGS_BYTE, (byte) (1 << Flags.IS_EMPTY.ordinal()));
+    setRebuildThreshold();
+  }
+
+  /**
```

Comment:
Is this javadoc necessary?  Generally, but not always, you can rely on the javadoc of the parent interface or abstract. If it is necessary, ignore this comment.  There are several like this.